rsconf/component/sirepo.py

```diff
@@ -271,16 +269,26 @@ def _install_dirs_and_files(self):
             )
 
     def _jupyterhublogin(self, z):
-        z.jupyterhub_enabled = "jupyterhublogin" in set(
-            self.j2_ctx.sirepo.feature_config.get("default_proprietary_sim_types", []),
-        ).union(set(self.j2_ctx.sirepo.feature_config.get("moderated_sim_types", [])))
+        z.jupyterhub_enabled = "jupyterhublogin" in self._in_sim_types(
+            "default_proprietary_sim_types", "moderated_sim_types"
+        )
         if not z.jupyterhub_enabled:
             return
         self.__docker_vols.append(z.sim_api.jupyterhublogin.user_db_root_d)
         self._set_sirepo_config("sirepo_jupyterhub")
 
+    def _raydata(self):
+        if "raydata" in self._in_sim_types("sim_types", "moderated_sim_types"):
+            self._set_sirepo_config("raydata_scan_monitor")
+
     def _set_sirepo_config(self, component):
         self.buildt.require_component(component)
         c = self.buildt.get_component(component)
         c.sirepo_config(self)
         self.__docker_unit_enable_after.append(c.name)
+
+    def _in_sim_types(self, *args):
+        s = set()
+        for a in args:
+            s = s.union(set(self.j2_ctx.sirepo.feature_config.get(a, [])))
```

Comment:
This function should look for all "sym_type" values in feature_config. Creating a set is doing more work than necessary:

```py
default _in_sim_types(self, to_check):
for k, v in self.j2_ctx.sirepo.feature_config.items():
    if "sim_types" in k and to_check in v:
        return True
return False
```

Reply:
3685a2ea3618ee935808a36d50a01190afebbc43